code/game/gamemodes/clock_cult/clock_structures.dm

```diff
@@ -0,0 +1,755 @@
+//////////////////////////
+// CLOCKWORK STRUCTURES //
+//////////////////////////
+
+/obj/structure/clockwork
+	name = "meme structure"
+	desc = "Some frog or something, the fuck?"
+	var/clockwork_desc //Shown to servants when they examine
+	icon = 'icons/obj/clockwork_objects.dmi'
+	icon_state = "rare_pepe"
+	anchored = 1
+	density = 1
+	opacity = 0
+	var/max_health = 100 //All clockwork structures have health that can be removed via attacks
+	var/health = 100
+	var/takes_damage = TRUE //If the structure can be damaged
+	var/break_message = "<span class='warning'>The frog isn't a meme after all!</span>" //The message shown when a structure breaks
+	var/break_sound = 'sound/magic/clockwork/anima_fragment_death.ogg' //The sound played when a structure breaks
+	var/list/debris = list(/obj/item/clockwork/component/replicant_alloy) //Parts left behind when a structure breaks
+	var/construction_value = 0 //How much value the structure contributes to the overall "power" of the structures on the station
+
+/obj/structure/clockwork/New()
+	..()
+	clockwork_construction_value += construction_value
+	all_clockwork_objects += src
+
+/obj/structure/clockwork/Destroy()
+	clockwork_construction_value -= construction_value
+	all_clockwork_objects -= src
+	..()
+
+/obj/structure/clockwork/proc/destroyed()
+	if(!takes_damage)
+		return 0
+	for(var/obj/item/I in debris)
+		new I (get_turf(src))
+	visible_message(break_message)
+	playsound(src, break_sound, 50, 1)
+	qdel(src)
+	return 1
+
+/obj/structure/clockwork/ex_act(severity)
+	if(takes_damage)
+		switch(severity)
+			if(1)
+				health -= max_health * 0.7 //70% max health lost
+			if(2)
+				health -= max_health * 0.4 //40% max health lost
+			if(3)
+				if(prob(50))
+					health -= max_health * 0.1 //10% max health lost
+		if(health <= max_health * 0.1) //If there's less than 10% max health left, destroy it
+			destroyed()
+			qdel(src)
+
+/obj/structure/clockwork/examine(mob/user)
+	if((is_servant_of_ratvar(user) || isobserver(user)) && clockwork_desc)
+		desc = clockwork_desc
+	..()
+	desc = initial(desc)
+
+/obj/structure/clockwork/attackby(obj/item/I, mob/living/user, params)
+	if(user.a_intent == "harm" && user.canUseTopic(I) && I.force)
+		user.visible_message("<span class='warning'>[user] strikes [src] with [I]!</span>", "<span class='danger'>You strike [src] with [I]!</span>")
+		playsound(src, I.hitsound, 50, 1)
+		user.changeNext_move(CLICK_CD_MELEE)
+		user.do_attack_animation(src)
+		health = max(0, health - I.force)
+		if(!health)
+			destroyed()
+			return 0
```

Comment:
I refactored attackby so you could all attackable structures and machines would use the same code to handle this... Look at item_attack.dm . See how grille or windows handle it now.